lib/blockstore/level.js

```diff
@@ -0,0 +1,128 @@
+/*!
+ * blockstore/level.js - leveldb block store for bcoin
+ * Copyright (c) 2019, Braydon Fuller (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+const {isAbsolute, resolve} = require('path');
+const bdb = require('bdb');
+const assert = require('bsert');
+const AbstractBlockStore = require('./abstract');
+const layout = require('./layout');
+
+/**
+ * LevelDB Block Store
+ *
+ * @alias module:blockstore:LevelBlockStore
+ * @abstract
+ */
+
+class LevelBlockStore extends AbstractBlockStore {
+  /**
+   * Create a blockstore that stores blocks in LevelDB.
+   * @constructor
+   */
+
+  constructor(options) {
+    super();
+
+    assert(isAbsolute(options.location), 'Location not absolute.');
+
+    this.location = options.location;
+    this.db = bdb.create({
+      location: resolve(this.location, './index')
```

Comment:
Could we allow `memory: true` to be passed down here for in-memory backend? It might be useful for testing etc.

Reply:
Possibly, however testing cases can usually stub things as necessary. Tests could implement a stubbed BlockStore when needed.

Reply:
Resolved, and have also integrated the changes to chain.